core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -694,7 +692,7 @@ public PlanOptimizers(
                         ImmutableSet.of(
                                 new RemoveRedundantIdentityProjections(),
                                 new PushDownProjectionsFromPatternRecognition())),
-                new MetadataQueryOptimizer(metadata),
+                new MetadataQueryOptimizer(plannerContext),
```

Comment:
The choice which optimizers now take plannerContext and which continue to take metadata looks arbitrary. What's the rule?